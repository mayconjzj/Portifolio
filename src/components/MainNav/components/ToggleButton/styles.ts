import tw from 'tailwind-styled-components';

export const Content = tw.div`
  md:hidden
  flex
`;
