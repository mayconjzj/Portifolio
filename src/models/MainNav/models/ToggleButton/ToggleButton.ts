export type ToggleButtonProps = {
  isOpenMenu: boolean;
  handleToggleMenu: () => void;
};
