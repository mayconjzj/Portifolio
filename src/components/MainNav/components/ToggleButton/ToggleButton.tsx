'use client';

import { HiBars2, HiXMark } from 'react-icons/hi2';

import { useToggleMenu } from './hooks';
import * as S from './styles';

export const ToggleButton = () => {
  const { isOpenMenu, handleToggleMenu } = useToggleMenu();

  return (
    <S.Content onClick={handleToggleMenu}>
      {!isOpenMenu && <HiBars2 />}
      {isOpenMenu && <HiXMark />}
    </S.Content>
  );
};
