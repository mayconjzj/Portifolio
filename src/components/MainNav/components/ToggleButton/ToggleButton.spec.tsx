import { render } from '@testing-library/react';

import { ToggleButton } from './ToggleButton';

describe('ToggleButton', () => {
  it('should render the mainnav', () => {
    const { container } = render(
      <ToggleButton
        isOpenMenu={false}
        handleToggleMenu={() => {
          'Next test';
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
