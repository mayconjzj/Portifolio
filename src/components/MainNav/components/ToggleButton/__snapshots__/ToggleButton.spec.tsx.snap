// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton should render the mainnav 1`] = `
<div>
  <div
    class="md:hidden flex text-3xl"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3 9a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 9zm0 6.75a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
